File: Selenium/Project/Activity4.java
package project;

import org.testng.annotations.Test;

import io.github.bonigarcia.wdm.WebDriverManager;

import org.testng.annotations.BeforeClass;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.testng.annotations.AfterClass;

public class Activity4 {
	
	WebDriver driver;
	
	  @BeforeClass
	  public void beforeClass() {
		  
		  WebDriverManager.firefoxdriver().setup();
		  driver = new FirefoxDriver();
		  Actions actions = new Actions(driver);
		  driver.get("http://alchemy.hguy.co/orangehrm");
	  }
	  
  @Test
  public void addEmp() {
	  
	    WebElement username = driver.findElement(By.id("txtUsername"));
		WebElement password = driver.findElement(By.id("txtPassword"));
		username.sendKeys("orange");
		password.sendKeys("orangepassword123");
		WebElement loginButton = driver.findElement(By.name("Submit"));
		loginButton.click();
		WebElement pimId = driver.findElement(By.id("menu_pim_viewPimModule"));
		pimId.click();
		driver.findElement(By.id("btnAdd")).click();
		WebElement fullName = driver.findElement(By.id("firstName"));
		WebElement lastName = driver.findElement(By.id("lastName"));
		WebElement saveButton = driver.findElement(By.id("btnSave"));
		fullName.sendKeys("John");
		lastName.sendKeys("K");
		saveButton.click();
		driver.navigate().back();
		System.out.println("Title is : " +  driver.getTitle());
		driver.navigate().back();
		WebElement empName = driver.findElement(By.id("empsearch_employee_name_empName"));
		WebElement id = driver.findElement(By.id("empsearch_id"));
		empName.sendKeys(Keys.chord(Keys.CONTROL, "a", Keys.DELETE));
		id.clear();
		empName.sendKeys("John");
		WebElement searchButton = driver.findElement(By.id("searchBtn"));
		searchButton.click();
		/*
		 * Actions actions = null;
		 * actions.moveToElement(searchButton).click().build().perform();
		 */
  }
  
  @AfterClass
  public void afterClass() {
	  
	  driver.close();
  }

}

File: Selenium/Project/activity4.py
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver :
    Action = ActionChains(driver)
    driver.get("http://alchemy.hguy.co/orangehrm")

    username = driver.find_element(By.ID,"txtUsername")
    password = driver.find_element(By.ID,"txtPassword")

    username.send_keys("orange")
    password.send_keys("orangepassword123")

    loginButton = driver.find_element(By.NAME,"Submit")
    loginButton.click()

    pimId = driver.find_element(By.ID,"menu_pim_viewPimModule")
    pimId.click()

    addButton = driver.find_element(By.ID,"btnAdd")
    addButton.click()

    fullName = driver.find_element(By.ID,"firstName")
    lastName = driver.find_element(By.ID,"lastName")
    saveButton = driver.find_element(By.ID,"btnSave")

    fullName.send_keys("John")
    lastName.send_keys("K")
    saveButton.click()

    driver.back()
    print("Title is : ", driver.title)
    driver.back()

    empName = driver.find_element(By.ID,"empsearch_employee_name_empName")
    id = driver.find_element(By.ID,"empsearch_id")

    empName.send_keys("John")
    id.clear()

    searchButton = driver.find_element(By.ID,"searchBtn")
    searchButton.click()
    #ActionChains(driver).move_to_element(searchButton).click()